src/shared/psi-util.c

```diff
@@ -0,0 +1,115 @@
+        const char *t, *cline;
+        bool field_filled[] = { false, false, false, false };
+        int r;
+
+        assert(path);
+        assert(IN_SET(type, PRESSURE_TYPE_SOME, PRESSURE_TYPE_FULL));
+        assert(ret);
+
+        if (type == PRESSURE_TYPE_SOME)
+                t = "some";
+        else if (type == PRESSURE_TYPE_FULL)
+                t = "full";
+        else
+                return -EINVAL;
+
+        f = fopen(path, "re");
+        if (!f)
+                return -errno;
+
+        for (;;) {
+                _cleanup_free_ char *l = NULL;
+                char *w;
+
+                r = read_line(f, LONG_LINE_MAX, &l);
+                if (r < 0)
+                        return r;
+                if (r == 0)
+                        break;
+
+                w = first_word(l, t);
+                if (w) {
+                        line = TAKE_PTR(l);
+                        break;
+                }
+        }
+
+        if (!line)
+                return -ENODATA;
+
+        /* skip type string */
+        cline = line + 4;
+
+        /* extracts either avgX=Y.Z or total=X */
+        while ((r = extract_first_word(&cline, &word, NULL, 0)) > 0) {
+                _cleanup_free_ char *a = NULL, *b = NULL;
+                const char *cword;
+
+                /* extracts (avgX and Y.Z) or (total and X) */
+                cword = word;
+                r = extract_many_words(&cword, "=", 0, &a, &b, NULL);
+                word = mfree(word);
+                if (r < 0)
+                        return r;
+                if (r != 2)
+                        return -EINVAL;
+
+                if (streq(a, "avg10")) {
+                        field_filled[0] = true;
+                        r = parse_loadavg_fixed_point(b, &rp.avg10);
+                } else if (streq(a, "avg60")) {
+                        field_filled[1] = true;
+                        r = parse_loadavg_fixed_point(b, &rp.avg60);
+                } else if (streq(a, "avg300")) {
+                        field_filled[2] = true;
+                        r = parse_loadavg_fixed_point(b, &rp.avg300);
+                } else if (streq(a, "total")) {
+                        field_filled[3] = true;
+                        r = parse_time(b, &rp.total, 1);
+                } else
+                        continue;
+
+                if (r < 0)
+                        return r;
+        }
+
+        if (r < 0)
+                return r;
+
+        if (!field_filled[0] || !field_filled[1] || !field_filled[2] || !field_filled[3])
+                return -EINVAL;
+
+        *ret = rp;
+        return 0;
+}
+
+bool is_pressure_supported(void) {
+        const char *p;
+
+        FOREACH_STRING(p, PRESSURE_CPU_FILE, PRESSURE_IO_FILE, PRESSURE_MEM_FILE)
+                if (access(p, R_OK) < 0) {
+                        log_debug_errno(errno, "No read access on %s: %m", p);
+                        return false;
+                }
+
+        return true;
```

Comment:
hmm, i guess these files exist in containers too, but are not virtualized for containers, right? So maybe issue detect_container() and return false if we run in a container?

Reply:
But I think in cgroup2 pressure files are in containers even if /proc/pressure/* is not is containerized. Maybe detect_container() and checking if not unified?

Reply:
I don't understand why being in a container or not is relevant to this. I thought that because of the hierarchical nature of the unified hierarchy it would be OK to run oomd in containers too. The check seems OK as is.

Reply:
I guess since PSI is only supported for cgroup v2 and would work in containers due to the unified hierarchy like you mentioned, the container check wouldn't be useful.